### PR TITLE
Refactored order in connect your wallet screen

### PR DESCRIPTION
**Pull request by kacperoak**

### CODE REVIEW

**Review by Elyniss — CHANGES_REQUESTED** [review]
:)

**Review by jakubuid — APPROVED** [review]

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/RecentWalletsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.walletconnect.web3.modal.domain
+
+import android.content.SharedPreferences
+
+private const val RECENT_WALLET_ID = "w3m_recent_wallet_id"
+
+class RecentWalletsRepository(
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/RecentWalletsRepository.kt:7]: let's make this internal

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/usecase/GetRecentWalletUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.walletconnect.web3.modal.domain.usecase
+
+import com.walletconnect.web3.modal.domain.RecentWalletsRepository
+
+class GetRecentWalletUseCase(private val repository: RecentWalletsRepository) {
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/usecase/GetRecentWalletUseCase.kt:5]: internal

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/usecase/SaveRecentWalletUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.walletconnect.web3.modal.domain.usecase
+
+import com.walletconnect.web3.modal.domain.RecentWalletsRepository
+
+class SaveRecentWalletUseCase(
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/usecase/SaveRecentWalletUseCase.kt:5]: internal

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**